components/WalletProvider.tsx

```diff
@@ -78,11 +80,52 @@ export const WalletProvider = ({
     [address, disconnect]
   )
 
+  const changeNetwork = async () => {
+    if (window.ethereum.networkVersion !== chainId) {
+      try {
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: `0x${chainId}` }],
+        })
+        return true
+      } catch (err: any) {
```

Comment:
Our linter is going to be displeased with this `any` type

Reply:
I see, changed it!